src/backend/optimizer/plan/createplan.c

```diff
@@ -6717,7 +6721,8 @@ adjust_modifytable_flow(PlannerInfo *root, ModifyTable *node, List *is_split_upd
 				if (optimizer_replicated_table_insert &&
 					subplan->flow->flotype == FLOW_SINGLETON &&
 					subplan->flow->locustype == CdbLocusType_General &&
-					!contain_volatile_functions((Node *)subplan->targetlist))
+					!contain_volatile_functions((Node *)subplan->targetlist) &&
+					!contain_volatile_functions(subplan->qual))
```

Comment:
Why do you handle locus General in the same maner? Do you have cases for it too?

Reply:
> Why do you handle locus General in the same maner?

reverted